yarn-project/archiver/src/archiver/eth_log_handlers.ts

```diff
@@ -63,61 +64,116 @@ export function processCancelledL1ToL2MessagesLogs(
  * @param publicClient - The viem public client to use for transaction retrieval.
  * @param expectedL2BlockNumber - The next expected L2 block number.
  * @param logs - L2BlockProcessed logs.
+ * @returns - An array of tuples representing block metadata including the header, archive tree root, and associated l1 block number
  */
-export async function processBlockLogs(
+export async function processBlockMetadataLogs(
   publicClient: PublicClient,
   expectedL2BlockNumber: bigint,
   logs: Log<bigint, number, false, undefined, true, typeof RollupAbi, 'L2BlockProcessed'>[],
-): Promise<L2Block[]> {
-  const retrievedBlocks: L2Block[] = [];
+): Promise<[Header, AppendOnlyTreeSnapshot, bigint][]> {
+  const retrievedBlockMetadata: [Header, AppendOnlyTreeSnapshot, bigint][] = [];
   for (const log of logs) {
     const blockNum = log.args.blockNumber;
     if (blockNum !== expectedL2BlockNumber) {
       throw new Error('Block number mismatch. Expected: ' + expectedL2BlockNumber + ' but got: ' + blockNum + '.');
     }
     // TODO: Fetch blocks from calldata in parallel
-    const newBlock = await getBlockFromCallData(publicClient, log.transactionHash!, log.args.blockNumber);
-    newBlock.setL1BlockNumber(log.blockNumber!);
-    retrievedBlocks.push(newBlock);
+    const [header, archive] = await getBlockMetadataFromRollupTx(
+      publicClient,
+      log.transactionHash!,
+      log.args.blockNumber,
+    );
+
+    retrievedBlockMetadata.push([header, archive, log.blockNumber!]);
     expectedL2BlockNumber++;
   }
-  return retrievedBlocks;
+
+  return retrievedBlockMetadata;
+}
+
+export async function processBlockBodyLogs(
```

Comment:
Would call this processTxsPublishedLogs to keep it consistent with the naming in the contract.